File: 2025/Day01/Task2/main.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
// main.js
var fs = require("fs");
var ptr = 50;
var score = 0;
var data = fs.readFileSync('./2025/Day01/Task1/input', 'utf8');
var lines = data.split("\n");
lines.forEach(function (line) {
    var direction = line[0] == "R" ? 1 : -1;
    var max_step = parseInt(line.substring(1));
    for (var i = 0; i < max_step; ++i) {
        ptr += direction;
        if (ptr > 99 || ptr < 0)
            ptr = ptr > 99 ? 0 : 99;
        if (ptr == 0)
            ++score;
    }
});
console.log(score);

File: 2025/Day01/Task2/main.lua
-- main.lua
local io = require("io")
local file = io.open("./2025/Day01/input", "r")

-- vars
local ptr = 50
local code = 0

-- error handling
if not file then
    print("Invalid input file!")
    os.exit(84)
end

-- func
for line in file:lines() do
    local direction = string.sub(line, 1, 1) == "R" and 1 or -1
    local step = string.sub(line, 2, #line)

    for _ = 1, step do
        code = code + (ptr + direction == 0 and 1 or ptr + direction == 100 and 1 or 0)
        ptr = ptr + direction == 100 and 0 or ptr + direction == 0 and ptr + direction or ptr + direction < 0 and 99 or ptr + direction
    end
end

print(code)

io.close(file)